File: sp2mic-backend/sp2mic.WebAPI/Domain/Entities/DtoClasse.cs
namespace sp2mic.WebAPI.Domain.Entities;

public class DtoClasse
{
  public DtoClasse()
  {
    Atributos = new HashSet<Atributo>();
    EndpointsQueRetornamEssaClasse = new HashSet<Endpoint>();
    StoredProceduresQueRetornamEssaClasse = new HashSet<StoredProcedure>();
  }

  public DtoClasse (string noDtoClasse)
  {
    NoDtoClasse = noDtoClasse;
    Atributos = new HashSet<Atributo>();
    EndpointsQueRetornamEssaClasse = new HashSet<Endpoint>();
    StoredProceduresQueRetornamEssaClasse = new HashSet<StoredProcedure>();
  }

  public int Id {get; set;}
  public string NoDtoClasse {get; set;} = null!;
  //public int? IdMicrosservico {get; set;}
  //public virtual Microsservico? IdMicrosservicoNavigation {get; set;}
  public int IdStoredProcedure {get; set;}
  public virtual StoredProcedure IdStoredProcedureNavigation {get; set;} = null!;

  public virtual ICollection<Atributo> Atributos {get; set;}
  public virtual ICollection<Endpoint> EndpointsQueRetornamEssaClasse {get; set;}
  public virtual ICollection<StoredProcedure> StoredProceduresQueRetornamEssaClasse {get; set;}
}

File: sp2mic-frontend/apps/sp2mic/src/app/shared/models/dto-classe.ts
import { AtributoDto } from './atributo-dto';

export class DtoClasse {
  id: number;
  noDtoClasse: string;
  //noMicrosservico: string;
  //idMicrosservico: number;
  idStoredProcedure: number;
  noStoredProcedure: string;
  txDtoClasse: string;
  //idMicrosservicoNavigation: Microsservico;
  atributos: AtributoDto[];
}

export class DtoClassePaginado extends DtoClasse {
  page: number;
  linesPerPage: number;
  orderBy: string;
  direction: string;
}
